packages/dapp-svelte-wallet/api/src/lib-dehydrate.js

```diff
@@ -233,14 +233,27 @@ export const makeDehydrator = (initialUnnamedCount = 0) => {
     };
 
     const suggestPetname = (petname, val) => {
-      if (petnameToVal.has(petname)) {
-        return;
-      }
       if (valToPetname.has(val)) {
-        return;
+        return valToPetname.get(val);
+      }
+
+      // Find a unique petname since we didn't already have the value,
```

Comment:
Hmm, there is something about this that violates the concept of petnames. I think there should not be a merge of autogenerated aspects + user-chosen names. The unknown-* defaults are the right way to deal with something that really has no name and no other information. @erights, what should be done?

Reply:
I can accept that for the string-based petnames.  For the array-based "paths", though, we already have a petname for the first element, but the rest of the elements are entirely chosen by the dapp.  It doesn't matter if we suffix them, they are already strong enough.

What is the correct behaviour in the pure petname (i.e. non-array-based) case?  The only client that would do this is one that isn't coming from a dapp, i.e. that already fully controls the ag-solo on which this code is running.

Reply:
Correction: the dapp itself suggests a petname for the origin, which is used as the first component of all its paths.

Reply:
@katelynsills would you PTAL?  I believe I've addressed your concern by only auto-suffixing petnames that are actually paths, since that is the only code path used by approved Dapps.

Unapproved Dapps can't suggest anything, and the only ones that can suggest bare strings are privileged users.

Reply:
I'm fine merging this in for now, but I think the "actually paths" distinction isn't a distinction. We aren't using edge names which are dynamic, so this is actually in the user's petname space whether we keep the path or not.